spec/system/frontend/solidus_stripe/checkout_spec.rb

```diff
@@ -2,218 +2,302 @@
-    expect_payments_state(order, ['completed'], outstanding: 0)
-  end
+      context 'with a registered user' do
+        it 'creates a setup intent and successfully processes payment' do
+          successfully_creates_a_setup_intent(user: create(:user))
+          complete_order
+          expect_payment_intent_to_be_created_successfully
+          capture_last_valid_payment
+        end
 
-  [true, false].each do |skip_confirm|
-    it "completes as a registered user using payment intents #{skip_confirm || ' not'} skipping confirm" do
-      create(
-        :stripe_payment_method,
-        preferred_stripe_intents_flow: 'payment',
-        preferred_skip_confirmation_for_payment_intent: skip_confirm,
-      )
-      visit_payment_step(user: create(:user))
-      choose_new_stripe_payment
-      fill_stripe_form
+        it 'successfully reuses a previously saved card from the user\'s wallet' do
+          user = create(:user)
 
-      if skip_confirm
-        check "Agree to Terms of Service"
-        click_button("Save and Continue")
-        expect(page).to have_content("Your order has been processed successfully")
-      else
-        submit_payment
-        expect_payments_state(Spree::Order.last, ['pending'])
-        confirm_order
+          successfully_creates_a_setup_intent(user: user)
+          complete_order
+          expect_payment_intent_to_be_created_successfully
+          capture_last_valid_payment
+
+          visit_payment_step(user: user)
+          find_existing_payment_radio(user.wallet_payment_sources.first.id).choose
+          submit_payment
+          complete_order
+          expect_payment_intent_to_be_created_successfully
+          capture_last_valid_payment
+        end
+
+        it 'creates a setup intent with a 3D Secure card' do
+          visit_payment_step(user: create(:user))
+          choose_new_stripe_payment
+
+          # Fill in the Stripe payment form with a 3D Secure card
+          # This card requires authentication on all transactions,
+          # regardless of how the card is set up.
+          # See https://stripe.com/docs/testing#authentication-and-setup for more information on this card.
+          fill_stripe_form(number: '4000002760003184')
+
+          # Submit the payment form and authorize the 3D Secure payment
+          submit_payment
+          authorize_3d_secure_payment
+
+          # Expect the setup intent to be created successfully and complete the order
+          expect_setup_intent_to_be_created_successfully
+          complete_order
+          expect(page).to have_content('Your order has been processed successfully')
+
+          # Note that, in case of a Setup Intent, even if it is authorized, the payment intent
+          # cannot be captured because a required action is needed most of the times.
+          expect_payment_intent_to_be_created_with_required_action
+
+          # TODO: These cards cannot be reused as the required action is not handled.
```

Comment:
nit: a failing test preceded by a `pending` would be ok here too, not blocking for the approval but if it's super easy to add would be of great help for the next one needing to implement the solution.

Reply:
Great idea! 👏  I've made the changes:
https://github.com/solidusio/solidus_stripe/blob/759cb54b80ce518a1f97828d29aa4677630d07de/spec/system/frontend/solidus_stripe/checkout_spec.rb#L66

This is the result:
<img width="1550" alt="Screenshot 2023-03-15 at 19 19 28" src="https://user-images.githubusercontent.com/19948291/225406528-dc2bd31d-fe49-4c65-8024-76b64c46a615.png">

Let me know if you were considering adding the `pending` status in the header of the "it" block instead (With the inclusion of the message, it appeared to be more comprehensive to me 🤔).